src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -977,7 +973,8 @@ private bool ConsumeStringNextSegment()
                 _bytePositionInLine += localBuffer.Length;
             }
 
-            SequencePosition start = new SequencePosition(startPosition.GetObject(), startPosition.GetInteger() + startConsumed);
+            SequencePosition originalPosition = rollBackState._prevCurrentPosition;
+            SequencePosition start = new SequencePosition(originalPosition.GetObject(), originalPosition.GetInteger() + rollBackState._prevConsumed + 1);
```

Comment:
Consider either saving this value before the loop or making a method on rollBackState (presumably taking an offset to handle the string vs number relative start differential)

Reply:
I am going to make it a method to avoid unnecessarily creating a SequencePosition up front before we know we need it.